docs/FIRRTLAnnotations.md

```diff
@@ -657,7 +657,7 @@ This key allows the creation of a FIRRTL literal.
 
 The DataTapsAnnotation is a collection of all the data taps in a circuit. This
 will cause a data tap module to be emitted.  The DataTapsAnnotation implies
-DontTouchAnnotation.
+DontTouchAnnotation on the `blackbox` target.
```

Comment:
This is technically on all _keys_ and not on the blackBox target.  I.e., anything that is a key should not be removed from the circuit.

Reply:
Yeah, looks like you're right, I was reading the code wrong.  Looks like we are attaching the `DontTouchAnnotation` on to the [blackBox](https://github.com/llvm/circt/blob/main/lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp#L673), almost every blackbox port (targeted by a key), and some ModuleTargets, but upstream it's only the `ReferenceDataTapKey.source` that gets DontTouch (EmitDataTaps.scala:115)? 

```scala
  def dontTouches: Seq[ReferenceTarget] = {
    keys.collect {
      case key: ReferenceDataTapKey => key.source
    }
```

Any idea what would make the most sense to document here?

The View annotation doesn't document its don't touches on every GroundType, which I will also add:
```scala
// ViewAnnotation
  def collectDontTouches(tpe: AugmentedType): Seq[ReferenceTarget] = tpe match {
    case b: AugmentedBundleType =>
      b.elements.flatMap { field =>
        collectDontTouches(field.tpe)
      }
    case v: AugmentedVectorType => v.elements.flatMap(collectDontTouches(_))
    case g: AugmentedGroundType => Seq(g.ref)
    case other => Seq.empty
  }
```

Reply:
Updated change only mentions what SFC puts don't touch on (`ReferenceDataTapKey` and `AugmentedGeroundType`). I am not sure if I should mention `DataTapModuleSignalKey.module` adding DontTouch.  Most of the other `DontTouches` we add (to the blackbox+ports) seem like an implementation detail and not worth mentioning.   WDYT?

Reply:
Good catch. It looks we're being overzealous with dont touch relative to the SFC implementation. We may need to walk that back.